.github/workflows/auto-docs-preview.yml

```diff
@@ -0,0 +1,263 @@
+# GitHub Action workflow for automatically creating documentation preview links in PRs
+# This workflow addresses issue #3189 to automate the preview creation process
+
+name: Auto-generate Docs PR Preview
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+    paths:
+      - 'docs/content/**'
+      - 'docs/README.md'
+      - 'docs/mkdocs.yml'
+      - 'docs/overrides/**'
+      - 'docs/scripts/**'
+      - 'docs/requirements.txt'
+      - 'docs/main.py'
+      - 'SECURITY.md'
+      - 'CODE_OF_CONDUCT.md'
+      - 'GOVERNANCE.md'
+      - 'ONBOARDING.md'
```

Comment:
This is more complex than https://github.com/rishi-jat/kubestellar/blob/54ce7f5beedeb36e6debe303428960a12eef3ec0/.github/workflows/docs-gen-and-push.yml and omits itself.